ygot/gostruct.go

```diff
@@ -0,0 +1,51 @@
+package ygot
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/openconfig/goyang/pkg/yang"
+	"github.com/openconfig/ygot/util"
+)
+
+const (
+	// GoCompressedLeafAnnotation is the yang.Entry annotation name to
+	// indicate that a particular leaf entry has a sibling that is
+	// compressed out.
+	GoCompressedLeafAnnotation = "ygot-oc-compressed-leaf"
+)
+
+// PruneReadOnly removes branches or leaf nodes that contain "config false" data in-place.
+//
+// Note that the input GoStruct MUST NOT be read-only, since only anchored
```

Comment:
I mean compressed GoStructs generated using `PreferOperationalState`. I'll clarify this.

Reply:
Sorry scratch what I said. I meant here that the input node itself MUST not be `config false` because the function doesn't have a reference to the pointer to the GoStruct. In other words, the function can overwrite any descendant to the input GoStruct, but not itself. Maybe this is a defect and we should clear the GoStruct in this case?

Reply:
So I've changed the behaviour to allow calling `PruneConfigFalse` on a GoStruct that is itself `config false`. The behaviour here is that the GoStruct's leaves get erased (since they're all also config false).

Reply:
I'm a bit lost here. Perhaps we can talk in real-time about this?

Reading through, what I think I understood is that if you hand a parent `GoStruct` to this function that is `config false`, now we clear all its fields (sounds right, `config false` is inherited down the schema tree). Let's use the terminology `config false` rather than `read-only` throughout this code because honestly, I can't see us remembering exactly the nuances here, but `config false` is well defined in rfc6020.

Reply:
Yes that's right. It clears all its fields. The second unit test tests the case of the input GoStruct being entirely `config false`.

I've thoroughly changed the docstring for this function now. There is a difference in behaviour between compressed vs. uncompressed GoStructs. I think the current way makes sense as an option, but alternatively to reduce confusion maybe it makes sense to make this function only remove derived state for uncompressed GoStructs as well.